01projects/sec04CMakeHelloWorld.md

```diff
@@ -0,0 +1,237 @@
+This is achieved by using the [`add_subdirectory()`](https://cmake.org/cmake/help/latest/command/add_subdirectory.html) command.
+
+``` cmake
+add_subdirectory(src)
+```
+
+In the above line, we instruct CMake to process the `CMakeLists.txt` located within the `src/` directory.
+These list files can be nested quite deeply, and at each depth level, we can have as many 'sibling' folders with one `CMakeLists.txt` per folder that describes the logical program unit to be built.
+
+After processing all nested list files, control returns to the top level `CMakeLists.txt`.
+Since the above line was the last in this file, the configuration process is completed.
+We now focus on the build details that is to be described in `src/CMakeLists.txt`.
+
+### Contents of `src/CMakeLists.txt`
+
+Within this 'leaf' `CMakeLists.txt` file, we describe how the `hello_world` executable is to be built.
+This is considered best practice, i.e. to have a `CMakeLists.txt` as close to the relevant source files for each modular functionality within the project.
+This file can see and use all the properties, variables, functions and macros defined at any `CMakeLists.txt` in its parent scope (in this case, all definitions from the top-level `CMakeLists.txt` file are visible here i.e. they are in the 'scope' of this leaf file).
+
+#### Targets and usage requirements
+
+We like CMake to build for us an executable called 'hello_world'.
+In this case, `hello_world` is a CMake 'target' (or a CMake 'executable target' to be precise).
+Targets (not variables) are the first-class citizens in modern CMake language.
+Hence, the relationships between various constituents (i.e. executables, source files, header files, internal libraries, other third-party code/files, system dependencies etc.) is to be described in a target-centric way within each `CMakeLists.txt` file.
+
+The behaviour of targets, their features and their various properties can be set in three ways:
+- only on/for/affecting itself (using the PRIVATE keyword)
+- only on/for/affecting other targets that consume this current target (using the INTERFACE keyword)
+- on/for/affecting itself as well as other targets that consume this current target (using the INTERFACE keyword)
+
+We recognise that this is quite a confusing aspect for newcomers to CMake, but is one of the key things to understand how targets and their dependencies interact for a given build.
+The only way to get better comprehend this critical component is by practicing these ideas during the first few weeks of the course.
+
+The following matrix may help slightly, and we may choose to refer to this at times throughout the course:
+
+Keyword selection matrix (Who needs/uses a dependency?)
+
+|                | Other (consuming targets) |
+|----------------|------------|---------|
+| **Current target** | **YES**        | **NO**      |
+| **YES**            | PUBLIC     | PRIVATE |
+| **NO**             | INTERFACE  | N/A     |
```

Comment:
I would recommend calling it a "Table" rather than a "Matrix". 

I'm not sure if it's my browser viewing the md file but this table is not rendering for me... it might be okay on the website build though.

Reply:
@mmcleod89 Fixed. Calling it a "table" now. The rendering issue seems to be fixed on the jekyll rendered pages